src/masternodes/rpc_proposals.cpp

```diff
@@ -474,9 +475,27 @@ UniValue votegov(const JSONRPCRequest &request) {
             throw JSONRPCError(RPC_INVALID_PARAMETER,
                                strprintf("Proposal <%s> is not in voting period", propId.GetHex()));
         }
+        if (id.length() == 64) {
+            mnId = ParseHashV(id, "masternodeId");
+        } else {
+            CTxDestination dest = DecodeDestination(id);
+            if (!IsValidDestination(dest)) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER,
+                                    strprintf("The masternode id or address is not valid: %s", id));
+            }
+            const CKeyID ckeyId = dest.index() == PKHashType ? CKeyID(std::get<PKHash>(dest)) : CKeyID(std::get<WitnessV0KeyHash>(dest));
+            auto masterNodeIdByOwner = view.GetMasternodeIdByOwner(ckeyId);
+            if (!masterNodeIdByOwner) {
+                auto masterNodeIdByOperator = view.GetMasternodeIdByOperator(ckeyId);
```

Comment:
Check that `masterNodeIdByOperator` is valid or you will get a `std::bad_optional_access` thrown on `masterNodeIdByOperator.value()`. I'd structure it like the following:

```
if (auto masterNodeIdByOwner = view.GetMasternodeIdByOwner(ckeyId)) {
    mnId = masterNodeIdByOwner.value();
} else if (auto masterNodeIdByOperator = view.GetMasternodeIdByOperator(ckeyId)) {
    mnId = masterNodeIdByOperator.value();
}
```